content/blog/2024-07-31-valkey-8-0-0-rc1.md

```diff
@@ -0,0 +1,126 @@
+I/O threads to operate concurrently. This enhancement significantly increases
+throughput, achieving up to 1.2 million Queries Per Second (QPS) on AWS's r7g
+platform, compared to the previous limit of 380K QPS.
+
+Valkey 8.0 focuses on intelligent optimization, not just raw speed. By fine-
+tuning CPU core utilization, minimizing command latency, and incorporating
+Memory Access Amortization (MAA) to streamline dictionary operations, Valkey
+8.0 delivers an in-memory data store that is both efficient and responsive.
+
+- **Asynchronous I/O Threading**: Enables parallel processing of commands and
+  I/O operations, maximizing throughput and minimizing bottlenecks.
+- **Optimized CPU Core Utilization**: Distributes I/O tasks across multiple
+  cores, reducing idle time and maximizing computational power.
+- **Memory Access Amortization (MAA)**: Optimizes memory access patterns by
+  prefetching frequently accessed data to minimize CPU cache misses, reducing
+  memory accesses required for dictionary operations.
+
+## Reliability
+
+Cluster scaling operations, particularly re-sharding and slot migrations, have
+historically been delicate. Valkey 8.0 improves reliability and minimizes
+disruptions with the following enhancements:
+
+- **Automatic Failover for Empty Shards**: New shards that start empty, owning
+  no slots, now benefit from automatic failover. This ensures high availability
+  from the start of the scaling process.
+- **Replication of Slot Migration States**: All `CLUSTER SETSLOT` commands are
+  now replicated synchronously on replicas before execution on the primary. This
+  guarantees a consistent slot migration state across the shard, preventing data
+  loss or inconsistencies if the primary fails.
+- **Logical Link Recovery**: In the event of a failover, Valkey 8.0 automatically
+  repairs the logical links between source and target nodes. This ensures requests
+  are correctly routed to the new primary in the target shard, maintaining cluster
+  integrity and availability.
+- **Seamless Support for New Replicas**: New replicas joining a shard mid-migration
+  will automatically inherit the correct slot migration state, ensuring immediate
+  consistency.
+
+## Replication
+
+Valkey 8.0 introduces a dual-channel replication scheme, allowing the RDB and
+the replica backlog to be transferred simultaneously, accelerating synchronization.
+
+- **Reduced Memory Load**: By streaming replication data to the replica during
+  the full sync, the primary node experiences significantly less memory pressure.
+  The replica now manages the Client Output Buffer (COB) tracking, reducing the
+  likelihood of COB overruns and enabling larger COB sizes on the replica side.
+- **Lighter Main Process Load**: A dedicated connection for RDB transfer frees
+  the primary's main process from handling this data, allowing it to focus on
+  client queries and improving overall responsiveness.
```

Comment:
```suggestion
- **Reduced Parent Process Load**: A dedicated connection for RDB transfer frees
  the primary's parent process from handling this data, allowing it to focus on
  client queries and improving overall responsiveness.
```
Typically parent is used instead of main process. But, not sure people will necessarily understand what this means. I almost just wonder if we should say "Reduced Primary load".